src/code/engine_library/addDependentPopulationParameter.m

```diff
@@ -0,0 +1,253 @@
+%   parPaths (cellarray of strings) pathnames of parameter
+%   parValues ( double array) values of parameter
+%   success (boolean) if false error has occured.
+%
+% Outputs:
+%   parPaths (cellarray of strings) pathnames of parameter
+%   parValues ( double array) values of parameter
+
+% Open Systems Pharmacology Suite;  http://open-systems-pharmacology.org
+
+success = true;
+
+[parPaths,parValues,successTmp] = addBSA(WSettings,parPaths,parValues);
+success = success & successTmp;
+
+%% add GFR
+[parPaths,parValues,successTmp] = addGFR(parPaths,parValues);
+success = success & successTmp;
+
+%% add GFR Norm
+[parPaths,parValues,successTmp] = addGFRBSANorm(parPaths,parValues);
+success = success & successTmp;
+
+%% add fraction unbounds
+[parPaths,parValues,successTmp] = addFU(parPaths,parValues);
+success = success & successTmp;
+
+return
+
+function [parPaths,parValues,success] = addFU(parPaths,parValues) 
+
+success = true;
+nPar = length(parPaths);
+    
+try
+    % %% fraction unbounds
+    [albuminOntgenyFactor,success] = getPopulationParameter(parPaths,parValues,'Organism|Ontogeny factor (albumin)',success);
+        
+    [agpOntogenyFactor,success] = getPopulationParameter(parPaths,parValues,'Organism|Ontogeny factor (alpha1-acid glycoprotein)',success);
+
+    if success
+        % get name of drugs with a positive fraction unbound
+        [~,desc] = existsParameter('*Fraction unbound (plasma, reference value)',1,'parametertype','readonly');
+        tmpStruct = cell2struct(desc(2:end,:),desc(1,:),2);
+        jj = [tmpStruct.Value] > 0;
+        for iJ = find(jj)
+            fu_ref = tmpStruct(iJ).Value;
+            tmp = regexp(tmpStruct(iJ).Path,'\|','split');
+            drug = tmp{2};
+            bindingPartner = getParameter(strjoin([tmp(1:2),{'Plasma protein binding partner'}],'|'),1,'parametertype','readonly');
+            switch bindingPartner
+                case 1
+                    FU =  1 ./ (1 + albuminOntgenyFactor .* (1 - fu_ref) ./ fu_ref);
+                case 2
+                    FU =  1 ./ (1 + agpOntogenyFactor .* (1 - fu_ref) ./ fu_ref);
+                otherwise
+                    FU = fu_ref ;
+            end
+            
+            parPaths{end+1} = strjoin({'DependentParameter',drug,'Fraction unbound (plasma)'},'|'); %#ok<AGROW>
+            parValues(:,end+1) = FU.*100; %#ok<AGROW>
+            
+        end
+        writeToReportLog('INFO','fraction unbound was added to population',false);
+    end
+catch
+    success = false;
+end
+
+if ~success
+    writeToReportLog('WARNING','It was not possible to add fraction unbound to population',false);
+    parPaths = parPaths(1:nPar);
+    parValues = parValues(:,1:nPar);
+end
+
+return
+
+function [value,success] = getPopulationParameter(popPaths,popValues,parPath,success)
+
+% initialize return value
+value = nan;
+
+% if it failed before this call return
+if ~success
+    return
+end
+
```

Comment:
It's OK, however a log message will be written only for the **_first_** missing parameter

Reply:
The first is enough. If one parameter is missing, that population parameter cant be calculated.